clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/ClusteredStore.java

```diff
@@ -87,15 +73,17 @@ private ClusteredStore(final OperationsCodec<K, V> codec, final ChainResolver<K,
   @Override
   public ValueHolder<V> get(final K key) throws StoreAccessException {
     Chain chain = storeProxy.get(key.hashCode());
-    ResolvedChain<K> resolvedChain = resolver.resolve(chain, key);
+    V value = null;
+    if(!chain.isEmpty()) {
+      ResolvedChain<K, V> resolvedChain = resolver.resolve(chain, key);
 
-    Chain compactedChain = resolvedChain.getCompactedChain();
-    storeProxy.replaceAtHead(key.hashCode(), chain, compactedChain);
+      Chain compactedChain = resolvedChain.getCompactedChain();
+      storeProxy.replaceAtHead(key.hashCode(), chain, compactedChain);
 
-    Operation<K> resolvedOperation = resolvedChain.getResolvedOperation(key);
-    V value = null;
-    if(resolvedOperation != null && resolvedOperation instanceof KeyValueOperation) {
-      value = ((KeyValueOperation<K, V>)resolvedOperation).getValue();
+      Operation<K, V> resolvedOperation = resolvedChain.getResolvedOperation(key);
+      if(resolvedOperation != null) {
+        value = resolvedOperation.getValue();
+      }
     }
     return new ClusteredValueHolder<V>(value);
```

Comment:
With your changes, this is no longer correct. A `ValueHolder` should never contain a `null`. You need to return `null` instead.